### DISCUSSION TITLE
storing Image

### DISCUSSION THREAD

**Discussion by Usama-Nasir000** [isar/isar#1535] [Help]
How can i store image in isar after Uint8List is not supported

**Comment by ndelanou** [comment]
You could hack this by converting your `Uint8List` and storing it as a `String`:
```dart
final encodedImage = String.fromCharCodes(imageUint8List);
```

**Comment by Cierra-Runis** [comment]
> How can i store image in isar after Uint8List is not supported

v4 support `List<int>`

**Comment by Jtplouffe** [comment]
You can use `List<byte>`, but I would strongly recommend against storing files inside the database.
It is almost always better to store them directly on disk, and to store the url to the file in the database.

**Reply by Cierra-Runis** [reply]
I am curiosity in why we shouldn't store it in database.

**Reply by Jtplouffe** [reply]
There are multiple reasons, but the main one is performance.
Storing files in the database will increase the database size, and will impact query performance.
By storing a file in the database, you are essentially storing a file inside of the database file.
Filesystems are designed to handle blob of data, whereas databases like Isar are not.

And, although it does not apply in this case since Isar is an embedded database, disk space for database is usually more expensive than disk space in a bucket.
